pilot/pkg/networking/plugin/authz/rbac_test.go

```diff
@@ -529,3 +529,190 @@ func TestConvertRbacRulesToFilterConfig(t *testing.T) {
+		{
+			ConfigMeta: model.ConfigMeta{Name: "service-role-1"},
+			Spec: &rbacproto.ServiceRole{
+				Rules: []*rbacproto.AccessRule{
+					{
+						Services: []string{"service"},
+						Methods:  []string{"GET"},
+					},
+				},
+			},
+		},
+		{
+			ConfigMeta: model.ConfigMeta{Name: "service-role-2"},
+			Spec: &rbacproto.ServiceRole{
+				Mode: rbacproto.EnforcementMode_PERMISSIVE,
+				Rules: []*rbacproto.AccessRule{
+					{
+						Services: []string{"service"},
+						Methods:  []string{"POST"},
+					},
+				},
+			},
+		},
+	}
+	bindings := []model.Config{
+		{
+			ConfigMeta: model.ConfigMeta{Name: "service-role-binding-1"},
+			Spec: &rbacproto.ServiceRoleBinding{
+				Subjects: []*rbacproto.Subject{
+					{
+						User: "user1",
+					},
+				},
+				RoleRef: &rbacproto.RoleRef{
+					Kind: "ServiceRole",
+					Name: "service-role-1",
+				},
+			},
+		},
+		{
+			ConfigMeta: model.ConfigMeta{Name: "service-role-binding-2"},
+			Spec: &rbacproto.ServiceRoleBinding{
+				Mode: rbacproto.EnforcementMode_PERMISSIVE,
+				Subjects: []*rbacproto.Subject{
+					{
+						User: "user2",
+					},
+				},
+				RoleRef: &rbacproto.RoleRef{
+					Kind: "ServiceRole",
+					Name: "service-role-1",
+				},
+			},
+		},
+		{
+			ConfigMeta: model.ConfigMeta{Name: "service-role-binding-3"},
+			Spec: &rbacproto.ServiceRoleBinding{
+				Subjects: []*rbacproto.Subject{
+					{
+						User: "user3",
+					},
+				},
+				RoleRef: &rbacproto.RoleRef{
+					Kind: "ServiceRole",
+					Name: "service-role-2",
+				},
+			},
+		},
+		{
+			ConfigMeta: model.ConfigMeta{Name: "service-role-binding-4"},
+			Spec: &rbacproto.ServiceRoleBinding{
+				Mode: rbacproto.EnforcementMode_PERMISSIVE,
+				Subjects: []*rbacproto.Subject{
+					{
+						User: "user4",
+					},
+				},
+				RoleRef: &rbacproto.RoleRef{
+					Kind: "ServiceRole",
+					Name: "service-role-2",
+				},
+			},
+		},
+	}
+
+	enforcePolicy1 := &policy.Policy{
+		Permissions: []*policy.Permission{generatePermission(":method", "GET")},
+		Principals:  []*policy.Principal{generatePrincipal("user1")},
+	}
+
+	permissivePolicy1 := &policy.Policy{
+		Permissions: []*policy.Permission{generatePermission(":method", "GET")},
+		Principals:  []*policy.Principal{generatePrincipal("user2")},
+	}
+
+	permissivePolicy2 := &policy.Policy{
+		Permissions: []*policy.Permission{generatePermission(":method", "POST")},
+		Principals:  []*policy.Principal{generatePrincipal("user3"), generatePrincipal("user4")},
+	}
+
+	enforcedRbac := &policy.RBAC{
+		Action: policy.RBAC_ALLOW,
+		Policies: map[string]*policy.Policy{
+			"service-role-1": enforcePolicy1,
+		},
+	}
+	permissiveRbac := &policy.RBAC{
+		Action: policy.RBAC_ALLOW,
+		Policies: map[string]*policy.Policy{
+			"service-role-1": permissivePolicy1,
+			"service-role-2": permissivePolicy2,
+		},
+	}
+	testCases := []struct {
+		name           string
+		service        string
+		rbac           *policy.RBAC
+		permissiveRbac *policy.RBAC
```

Comment:
probably keep the name consistent with output, e.g shadowRbac.

Reply:
renamed.